data/mods/ssb/conditions.ts

```diff
@@ -1,1850 +1,2238 @@
-export const Conditions: {[k: string]: ModdedConditionData} = {
+import {FS} from '../../../lib/fs';
+import {toID} from '../../../sim/dex-data';
+
+// Used in many abilities, placed here to reduce the number of updates needed and to reduce the chance of errors
+const STRONG_WEATHERS = ['desolateland', 'primordialsea', 'deltastream', 'heavyhailstorm', 'winterhail'];
+
+// Similar to User.usergroups. Cannot import here due to users.ts requiring Chat
+// This also acts as a cache, meaning ranks will only update when a hotpatch/restart occurs
+const usergroups: {[userid: string]: string} = {};
+const usergroupData = FS('config/usergroups.csv').readIfExistsSync().split('\n');
+for (const row of usergroupData) {
+	if (!toID(row)) continue;
+
+	const cells = row.split(',');
+	usergroups[toID(cells[0])] = cells[1] || ' ';
```

Comment:
```suggestion
	const cells = row.split(',');
	if (cells.length !== 2) return;
	usergroups[toID(cells[0])] = cells[1] || ' ';
```

Reply:
This ensures that you don't set `usergroups[undefined]` if the file is corrupted, which has happened before. (Alternatively, you could throw an error that you can nag the sysadmins about :P)

Reply:
@HoeenCoder you are needed here

Reply:
> @HoeenCoder you are needed here

:[

Reply:
To be clear they pined me because I wrote this along with alot of the core engine work for SSB.

I'd throw an error instead yes.
```suggestion
	const cells = row.split(',');
	if (cells.length !== 2) throw new Error(`Invalid entry when parsing usergroups.csv`);
	usergroups[toID(cells[0])] = cells[1] || ' ';
```